### PR TITLE
Server firebase files

### PR DESCRIPTION
**Pull request by matuella**
Adds core firebase files.

Documentation on these (including scripts and folder structure) will be added in a future PR.

### CODE REVIEW

File: firebase/firebase.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "functions": {
+    "source": "src/functions",
+    "predeploy": [
+      "npm --prefix \"$PROJECT_DIR\" run functions-build"
+    ]
+  },
+  "emulators": {
+    "storage": {
+      "port": 9199
+    },
```
Comment (ggirotto) [inline firebase/firebase.json:18]: Firebase Storage emulator was launched recently and still have a lot of bugs. I would remove it while it's not stable

File: firebase/firebase.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ]
+  },
+  "functions": {
+    "source": "src/functions",
+    "predeploy": [
+      "npm --prefix \"$PROJECT_DIR\" run functions-build"
+    ]
+  },
+  "emulators": {
+    "storage": {
```
Comment (ggirotto) [inline firebase/firebase.json:16]: Firebase Storage emulator was launched recently and still have a lot of bugs. I would remove it while it's not stable

File: firebase/firebase.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      "port": 9199
+    },
+    "auth": {
+      "port": 9099
+    },
+    "pubsub": {
+      "port": 8085
+    },
```
Comment (ggirotto) [inline firebase/firebase.json:24]: Why adding pubsub if we don't use it?

File: firebase/firebase.json (hunk 2 above)
Comment (ggirotto) [reply firebase/firebase.json:16]: Also, we don't use Storage right?

Comment (matuella) [reply firebase/firebase.json:16]: I've checked their docs and it seems pretty good all around (even `@firebase/rules-unit-testing` received a significant update with these recent changes). Anyways, these are just standard ports, having these here or not won't make any difference - anyone can start a emulator locally if they wanted to, although all scripts in `package.json` only emulate the specific services that we are planning to use, which are `firestore` and `functions`

File: firebase/firebase.json (hunk 3 above)
Comment (matuella) [reply firebase/firebase.json:24]: The comment above address your question as well.

**Review by ggirotto — APPROVED** [review]

**Comment by matuella** [comment]
All updates made here will be merged in the last PR (#149) of this nested chain of PRs.

**Pull request closed without merging**